Cannot find Unity 2019.2 installation on Windows

I can install it, but once it's actually installed I can't use it.  I believe it's related to Unity 2019.2's lack of an `ivy.xml`, which [this similar project is also running into](https://github.com/DragonBox/u3d/issues/367).

This is how I'm invoking it:

```powershell
Install-Module UnitySetup -Force -Verbose

Find-UnitySetupInstaller -Version 2019.2.2f1 -Components 'Windows' -Verbose | Install-UnitySetupInstance -Verbose

Get-UnitySetupInstance -Verbose
```

Here's the relevant part of the log for my GitHub Actions pipeline:

```
2019-08-30T22:06:01.1276722Z ##[group]Run ci/windows/install.ps1
2019-08-30T22:06:01.1277251Z [36;1mci/windows/install.ps1[0m
2019-08-30T22:06:01.3416527Z shell: C:\Program Files\PowerShell\6\pwsh.EXE -command ". '{0}'"
2019-08-30T22:06:01.3417035Z ##[endgroup]
2019-08-30T22:06:10.7166181Z Installing UnitySetup
2019-08-30T22:06:18.7384029Z VERBOSE: Acquiring providers for assembly: C:\program files\powershell\6\Modules\PackageManagement\coreclr\netstandard2.0\Microsoft.PackageManagement.MetaProvider.PowerShell.dll
2019-08-30T22:06:18.7401058Z VERBOSE: Acquiring providers for assembly: C:\program files\powershell\6\Modules\PackageManagement\coreclr\netstandard2.0\Microsoft.PackageManagement.ArchiverProviders.dll
2019-08-30T22:06:18.8413548Z VERBOSE: Acquiring providers for assembly: C:\program files\powershell\6\Modules\PackageManagement\coreclr\netstandard2.0\Microsoft.PackageManagement.CoreProviders.dll
2019-08-30T22:06:18.8876010Z VERBOSE: Acquiring providers for assembly: C:\program files\powershell\6\Modules\PackageManagement\coreclr\netstandard2.0\Microsoft.PackageManagement.NuGetProvider.dll
2019-08-30T22:06:19.9589522Z VERBOSE: Using the provider 'PowerShellGet' for searching packages.
2019-08-30T22:06:20.7741530Z VERBOSE: The -Repository parameter was not specified.  PowerShellGet will use all of the registered repositories.
2019-08-30T22:06:20.7851477Z VERBOSE: Getting the provider object for the PackageManagement Provider 'NuGet'.
2019-08-30T22:06:20.7911241Z VERBOSE: The specified Location is 'https://www.powershellgallery.com/api/v2' and PackageManagementProvider is 'NuGet'.
2019-08-30T22:06:21.2903530Z VERBOSE: Searching repository 'https://www.powershellgallery.com/api/v2/FindPackagesById()?id='UnitySetup'' for ''.
2019-08-30T22:06:22.5347533Z VERBOSE: Total package yield:'1' for the specified package 'UnitySetup'.
2019-08-30T22:06:22.8004680Z VERBOSE: Performing the operation "Install-Module" on target "Version '5.1.126' of module 'UnitySetup'".
2019-08-30T22:06:22.8985803Z VERBOSE: The installation scope is specified to be 'CurrentUser'.
2019-08-30T22:06:22.9012159Z VERBOSE: The specified module will be installed in 'C:\Users\runneradmin\Documents\PowerShell\Modules'.
2019-08-30T22:06:22.9234379Z VERBOSE: The specified Location is 'NuGet' and PackageManagementProvider is 'NuGet'.
2019-08-30T22:06:22.9240188Z VERBOSE: Downloading module 'UnitySetup' with version '5.1.126' from the repository 'https://www.powershellgallery.com/api/v2'.
2019-08-30T22:06:22.9709940Z VERBOSE: Searching repository 'https://www.powershellgallery.com/api/v2/FindPackagesById()?id='UnitySetup'' for ''.
2019-08-30T22:06:23.1538996Z VERBOSE: Searching repository 'https://www.powershellgallery.com/api/v2/FindPackagesById()?id='powershell-yaml'' for ''.
2019-08-30T22:06:23.1950254Z VERBOSE: InstallPackage' - name='powershell-yaml', version='0.4.0',destination='C:\Users\runneradmin\AppData\Local\Temp\1732613849'
2019-08-30T22:06:23.1967149Z VERBOSE: DownloadPackage' - name='powershell-yaml', version='0.4.0',destination='C:\Users\runneradmin\AppData\Local\Temp\1732613849\powershell-yaml.0.4.0\powershell-yaml.0.4.0.nupkg', uri='https://www.powershellgallery.com/api/v2/package/powershell-yaml/0.4.0'
2019-08-30T22:06:23.2005372Z VERBOSE: Downloading 'https://www.powershellgallery.com/api/v2/package/powershell-yaml/0.4.0'.
2019-08-30T22:06:23.3179935Z VERBOSE: Completed downloading 'https://www.powershellgallery.com/api/v2/package/powershell-yaml/0.4.0'.
2019-08-30T22:06:23.3193498Z VERBOSE: Completed downloading 'powershell-yaml'.
2019-08-30T22:06:23.3240906Z VERBOSE: InstallPackageLocal' - name='powershell-yaml', version='0.4.0',destination='C:\Users\runneradmin\AppData\Local\Temp\1732613849'
2019-08-30T22:06:23.6900448Z VERBOSE: InstallPackage' - name='UnitySetup', version='5.1.126',destination='C:\Users\runneradmin\AppData\Local\Temp\1732613849'
2019-08-30T22:06:23.6914109Z VERBOSE: DownloadPackage' - name='UnitySetup', version='5.1.126',destination='C:\Users\runneradmin\AppData\Local\Temp\1732613849\UnitySetup.5.1.126\UnitySetup.5.1.126.nupkg', uri='https://www.powershellgallery.com/api/v2/package/UnitySetup/5.1.126'
2019-08-30T22:06:23.6919393Z VERBOSE: Downloading 'https://www.powershellgallery.com/api/v2/package/UnitySetup/5.1.126'.
2019-08-30T22:06:23.7562238Z VERBOSE: Completed downloading 'https://www.powershellgallery.com/api/v2/package/UnitySetup/5.1.126'.
2019-08-30T22:06:23.7638218Z VERBOSE: Completed downloading 'UnitySetup'.
2019-08-30T22:06:23.7659527Z VERBOSE: InstallPackageLocal' - name='UnitySetup', version='5.1.126',destination='C:\Users\runneradmin\AppData\Local\Temp\1732613849'
2019-08-30T22:06:23.9424096Z VERBOSE: Validating the 'powershell-yaml' module contents under 'C:\Users\runneradmin\AppData\Local\Temp\1732613849\powershell-yaml.0.4.0' path.
2019-08-30T22:06:24.1591824Z VERBOSE: Test-ModuleManifest successfully validated the module manifest file 'C:\Users\runneradmin\AppData\Local\Temp\1732613849\powershell-yaml.0.4.0'.
2019-08-30T22:06:24.1619710Z VERBOSE: Validating the authenticode signature and publisher of the catalog file or module manifest file of the module 'powershell-yaml'.
2019-08-30T22:06:24.1870287Z VERBOSE: Catalog file 'powershell-yaml.cat' is not found in the contents of the module 'powershell-yaml' being installed.
2019-08-30T22:06:24.2600693Z VERBOSE: Checking for possible command collisions for the module 'powershell-yaml' commands.
2019-08-30T22:06:24.9559219Z VERBOSE: Installing the dependency module 'powershell-yaml' with version '0.4.0' for the module 'UnitySetup'.
2019-08-30T22:06:25.0541688Z VERBOSE: Module 'powershell-yaml' was installed successfully to path 'C:\Users\runneradmin\Documents\PowerShell\Modules\powershell-yaml\0.4.0'.
2019-08-30T22:06:25.1662207Z VERBOSE: Validating the 'UnitySetup' module contents under 'C:\Users\runneradmin\AppData\Local\Temp\1732613849\UnitySetup.5.1.126' path.
2019-08-30T22:06:25.5830963Z VERBOSE: Test-ModuleManifest successfully validated the module manifest file 'C:\Users\runneradmin\AppData\Local\Temp\1732613849\UnitySetup.5.1.126'.
2019-08-30T22:06:25.5849776Z VERBOSE: Validating the authenticode signature and publisher of the catalog file or module manifest file of the module 'UnitySetup'.
2019-08-30T22:06:25.5922904Z VERBOSE: Catalog file 'UnitySetup.cat' is not found in the contents of the module 'UnitySetup' being installed.
2019-08-30T22:06:25.6078633Z VERBOSE: Checking for possible command collisions for the module 'UnitySetup' commands.
2019-08-30T22:06:25.8210877Z VERBOSE: Module 'UnitySetup' was installed successfully to path 'C:\Users\runneradmin\Documents\PowerShell\Modules\UnitySetup\5.1.126'.
2019-08-30T22:06:25.8291862Z Done installing UnitySetup
2019-08-30T22:06:25.8306637Z Installing Unity
2019-08-30T22:06:26.5629864Z VERBOSE: GET https://unity3d.com/get-unity/download/archive with 0-byte payload
2019-08-30T22:06:26.6743117Z VERBOSE: received -byte response of content type text/html
2019-08-30T22:06:27.0877957Z VERBOSE: HEAD https://download.unity3d.com/download_unity/ab112815d860/Windows64EditorInstaller/UnitySetup64-2019.2.2f1.exe with 0-byte payload
2019-08-30T22:06:27.2408220Z VERBOSE: received 985188448-byte response of content type application/x-msdos-program
2019-08-30T22:06:27.3423410Z VERBOSE: Downloading https://download.unity3d.com/download_unity/ab112815d860/Windows64EditorInstaller/UnitySetup64-2019.2.2f1.exe to C:\Users\runneradmin\.unitysetup\Installers\Unity-2019.2.2f1\UnitySetup64-2019.2.2f1.exe
2019-08-30T22:08:57.0351715Z VERBOSE: Installing Windows Editor
2019-08-30T22:08:57.0426243Z VERBOSE: 08/30/2019 22:08:57: Installing Windows to C:\Program Files\Unity\Hub\Editor\2019.2.2f1.
2019-08-30T22:13:49.3631329Z VERBOSE: 08/30/2019 22:13:49: Succeeded.
2019-08-30T22:13:49.3653853Z Done installing Unity
2019-08-30T22:13:49.3917042Z VERBOSE: Creating UnitySetupInstance for C:\Program Files\Unity
2019-08-30T22:13:49.4163672Z VERBOSE: Attempting to find UnityVersion in C:\Program Files\Unity
2019-08-30T22:13:49.4165726Z VERBOSE: Looking for ivy.xml files under C:\Program Files\Unity\Editor\
2019-08-30T22:14:13.1999948Z VERBOSE: 	Looking for version in C:\Program Files\Unity\Editor\Data\PackageManager\Unity\PackageManager\2018.3.7\ivy.xml
2019-08-30T22:14:13.2162085Z VERBOSE: 	Found version!
2019-08-30T22:14:13.2426656Z 
2019-08-30T22:14:13.2460303Z VERBOSE: Creating UnitySetupInstance for C:\Program Files\Unity\Hub\Editor\2019.2.2f1
2019-08-30T22:14:13.2470168Z VERBOSE: Attempting to find UnityVersion in C:\Program Files\Unity\Hub\Editor\2019.2.2f1
2019-08-30T22:14:13.2508312Z VERBOSE: Looking for ivy.xml files under C:\Program Files\Unity\Hub\Editor\2019.2.2f1\Editor\
2019-08-30T22:14:13.6667767Z VERBOSE: 	Looking for version in C:\Program Files\Unity\Hub\Editor\2019.2.2f1\Editor\Data\UnityExtensions\Unity\UnityVR\ivy.xml
2019-08-30T22:14:13.6779769Z WARNING: Unable to find version for C:\Program Files\Unity\Hub\Editor\2019.2.2f1
2019-08-30T22:14:13.6812689Z Version    Components Path
2019-08-30T22:14:13.6814021Z -------    ---------- ----
2019-08-30T22:14:13.6815409Z 2018.3.7f1    Windows C:\Program Files\Unity
2019-08-30T22:14:13.6816106Z 
2019-08-30T22:14:13.7908604Z ##[group]Run ci/windows/activate.ps1
2019-08-30T22:14:13.7909200Z [36;1mci/windows/activate.ps1[0m
2019-08-30T22:14:13.7961898Z shell: C:\Program Files\PowerShell\6\pwsh.EXE -command ". '{0}'"
2019-08-30T22:14:13.7962632Z env:
2019-08-30T22:14:13.7963307Z   UNITY_USERNAME: ***
2019-08-30T22:14:13.7963772Z   UNITY_PASSWORD: ***
2019-08-30T22:14:13.7964193Z   UNITY_PLATFORM: Windows
2019-08-30T22:14:13.7964619Z ##[endgroup]
2019-08-30T22:14:14.3764150Z 
2019-08-30T22:14:14.3768824Z 
2019-08-30T22:14:14.3775969Z     Directory: D:\a\Chromavaders\Chromavaders
2019-08-30T22:14:14.3776403Z 
2019-08-30T22:14:14.3778736Z Mode                LastWriteTime         Length Name
2019-08-30T22:14:14.3779359Z ----                -------------         ------ ----
2019-08-30T22:14:14.3780603Z d-----         8/30/2019 10:14 PM                Windows
2019-08-30T22:14:17.9785588Z WARNING: Unable to find version for C:\Program Files\Unity\Hub\Editor\2019.2.2f1
2019-08-30T22:14:18.0619578Z Start-UnityEditor : Could not find Unity Editor for version 2019.2.2f1
2019-08-30T22:14:18.0622992Z At D:\a\Chromavaders\Chromavaders\ci\windows\activate.ps1:12 char:1
2019-08-30T22:14:18.0624206Z + Start-UnityEditor -Project "$env:GITHUB_WORKSPACE" -Credential $Crede ...
2019-08-30T22:14:18.0625006Z + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2019-08-30T22:14:18.0625780Z + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
2019-08-30T22:14:18.0626596Z + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,Start-UnityEditor
2019-08-30T22:14:18.0627263Z  
2019-08-30T22:14:18.0860009Z ##[error]Process completed with exit code 1.
```

--- a/UnitySetup/UnitySetup.psd1
+++ b/UnitySetup/UnitySetup.psd1
@@ -54,7 +54,7 @@
 
     # Modules that must be imported into the global environment prior to importing this module
     RequiredModules   = @(
-        @{ModuleName = "powershell-yaml"; ModuleVersion = "0.3"; Guid = "6a75a662-7f53-425a-9777-ee61284407da"}
+        @{ModuleName = "powershell-yaml"; ModuleVersion = "0.3"; Guid = "6a75a662-7f53-425a-9777-ee61284407da" }
     )
 
     # Assemblies that must be loaded prior to importing this module

--- a/UnitySetup/UnitySetup.psm1
+++ b/UnitySetup/UnitySetup.psm1
@@ -117,16 +117,16 @@ class UnityProjectInstance {
 
     UnityProjectInstance([string]$path) {
         $versionFile = [io.path]::Combine($path, "ProjectSettings\ProjectVersion.txt")
-        if (!(Test-Path $versionFile)) { throw "Path is not a Unity project: $path"}
+        if (!(Test-Path $versionFile)) { throw "Path is not a Unity project: $path" }
 
         $fileVersion = (Get-Content $versionFile -Raw | ConvertFrom-Yaml)['m_EditorVersion'];
-        if (!$fileVersion) { throw "Project is missing a version in: $versionFile"}
+        if (!$fileVersion) { throw "Project is missing a version in: $versionFile" }
 
         $projectSettingsFile = [io.path]::Combine($path, "ProjectSettings\ProjectSettings.asset")
-        if (!(Test-Path $projectSettingsFile)) { throw "Project is missing ProjectSettings.asset"}
+        if (!(Test-Path $projectSettingsFile)) { throw "Project is missing ProjectSettings.asset" }
 
         $prodName = ((Get-Content $projectSettingsFile -Raw | ConvertFrom-Yaml)['playerSettings'])['productName']
-        if (!$prodName) { throw "ProjectSettings is missing productName"}
+        if (!$prodName) { throw "ProjectSettings is missing productName" }
 
         $this.Path = $path
         $this.Version = $fileVersion
@@ -144,7 +144,7 @@ class UnityVersion : System.IComparable {
 
     [string] ToString() {
         $result = "$($this.Major).$($this.Minor).$($this.Revision)$($this.Release)$($this.Build)"
-        if ( $this.Suffix ) { $result += "-$($this.Suffix)"}
+        if ( $this.Suffix ) { $result += "-$($this.Suffix)" }
         return $result
     }
 
@@ -166,7 +166,7 @@ class UnityVersion : System.IComparable {
 
     [int] CompareTo([object]$obj) {
         if ($null -eq $obj) { return 1 }
-        if ($obj -isnot [UnityVersion]) { throw "Object is not a UnityVersion"}
+        if ($obj -isnot [UnityVersion]) { throw "Object is not a UnityVersion" }
 
         return [UnityVersion]::Compare($this, $obj)
     }
@@ -245,7 +245,7 @@ function Get-OperatingSystem {
 function Get-UnityEditor {
     [CmdletBinding()]
     param(
-        [ValidateScript( {Test-Path $_ -PathType Container} )]
+        [ValidateScript( { Test-Path $_ -PathType Container } )]
         [Parameter(Mandatory = $false, ValueFromPipeline = $true, Position = 0, ParameterSetName = "Path")]
         [string[]]$Path = $PWD,
 
@@ -426,27 +426,30 @@ function Find-UnitySetupInstaller {
             $searchPages += $patchPage
 
             $webResult = Invoke-WebRequest $patchPage -UseBasicParsing
-            $searchPages += $webResult.Links | Where-Object {
-                $_.href -match "\/unity\/qa\/patch-releases\?version=$($Version.Major)\.$($Version.Minor)&page=(\d+)" -and $Matches[1] -gt 1
-            } | ForEach-Object { "https://unity3d.com$($_.href)" }
+            $searchPages += $webResult.Links | 
+                Where-Object { $_.href -match "\/unity\/qa\/patch-releases\?version=$($Version.Major)\.$($Version.Minor)&page=(\d+)" -and $Matches[1] -gt 1 } | 
+                ForEach-Object { "https://unity3d.com$($_.href)" }
         }
     }
 
     foreach ($page in $searchPages) {
         try {
             $webResult = Invoke-WebRequest $page -UseBasicParsing
-            $prototypeLink = $webResult.Links | Select-Object -ExpandProperty href -ErrorAction SilentlyContinue | Where-Object {
-                $link = $_
+            $prototypeLink = $webResult.Links | 
+                Select-Object -ExpandProperty href -ErrorAction SilentlyContinue | 
+                Where-Object {
+                    $link = $_
 
-                foreach ( $installer in $installerTemplates.Keys ) {
-                    foreach ( $template in $installerTemplates[$installer] ) {
-                        if ( $link -like "*$template*" ) { return $true }
+                    foreach ( $installer in $installerTemplates.Keys ) {
+                        foreach ( $template in $installerTemplates[$installer] ) {
+                            if ( $link -like "*$template*" ) { return $true }
+                        }
                     }
-                }
 
-                return $false
+                    return $false
 
-            } | Select-Object -First 1
+                } | 
+                Select-Object -First 1
 
             if ($null -ne $prototypeLink) { break }
         }
@@ -465,10 +468,10 @@ function Find-UnitySetupInstaller {
         $knownBaseUrls = $linkComponents[0], $knownBaseUrls
     }
     else {
-        $knownBaseUrls = $knownBaseUrls | Sort-Object -Property @{ Expression = {[math]::Abs(($_.CompareTo($linkComponents[0])))}; Ascending = $true}
+        $knownBaseUrls = $knownBaseUrls | Sort-Object -Property @{ Expression = { [math]::Abs(($_.CompareTo($linkComponents[0]))) }; Ascending = $true }
     }
 
-    $installerTemplates.Keys |  Where-Object { $Components -band $_ } | ForEach-Object {
+    $installerTemplates.Keys | Where-Object { $Components -band $_ } | ForEach-Object {
         $templates = $installerTemplates.Item($_);
         $result = $null
         foreach ($template in $templates ) {
@@ -679,7 +682,7 @@ function Request-UnitySetupInstaller {
         $downloads = @()
 
         try {
-            $global:downloadData = [ordered]@{}
+            $global:downloadData = [ordered]@{ }
             $downloadIndex = 1
 
             $allInstallers | ForEach-Object {
@@ -934,7 +937,7 @@ function Install-UnitySetupInstance {
             $defaultInstallPath = $BasePath
         }
 
-        $versionInstallers = @{}
+        $versionInstallers = @{ }
     }
     process {
         # Sort each installer received from the pipe into versions
@@ -1170,7 +1173,7 @@ function Get-UnitySetupInstanceVersion {
     [CmdletBinding()]
     param(
         [ValidateNotNullOrEmpty()]
-        [ValidateScript( {Test-Path $_ -PathType Container})]
+        [ValidateScript( { Test-Path $_ -PathType Container })]
         [Parameter(Mandatory = $true, Position = 0)]
         [string]$Path
     )
@@ -1322,11 +1325,11 @@ function Get-UnityProjectInstance {
 
     Get-ChildItem @args |
         ForEach-Object {
-        $path = [io.path]::Combine($_.FullName, "ProjectVersion.txt")
-        if ( Test-Path $path ) {
-            [UnityProjectInstance]::new((Join-Path $_.FullName "..\" | Convert-Path))
+            $path = [io.path]::Combine($_.FullName, "ProjectVersion.txt")
+            if ( Test-Path $path ) {
+                [UnityProjectInstance]::new((Join-Path $_.FullName "..\" | Convert-Path))
+            }
         }
-    }
 }
 
 <#
@@ -1607,8 +1610,8 @@ function Start-UnityEditor {
             if ( $instanceArgs[$i] ) { $unityArgs += $instanceArgs[$i] }
 
             $actionString = "$editor $unityArgs"
-            if ( $Credential ) { $actionString += " -password (hidden)"}
-            if ( $Serial ) { $actionString += " -serial (hidden)"}
+            if ( $Credential ) { $actionString += " -password (hidden)" }
+            if ( $Serial ) { $actionString += " -serial (hidden)" }
 
             if (-not $PSCmdlet.ShouldProcess($actionString, "System.Diagnostics.Process.Start()")) {
                 continue

--- a/UnitySetup/UnitySetup.psm1
+++ b/UnitySetup/UnitySetup.psm1
@@ -1141,19 +1141,18 @@ function Get-UnitySetupInstance {
         }
     }
 
-    
-    $BasePath | Where-Object { Test-Path $_ -PathType Container } | 
-        Get-ChildItem -Directory | Where-Object { (Get-UnityEditor $_.FullName).Count -gt 0 } | 
+    Get-ChildItem -Path $BasePath -Directory -ErrorAction Ignore | 
+        Where-Object { (Get-UnityEditor $_.FullName).Count -gt 0 } | 
         ForEach-Object {
-        $path = $_.FullName
-        try {
-            Write-Verbose "Creating UnitySetupInstance for $path"
-            [UnitySetupInstance]::new($path)
+            $path = $_.FullName
+            try {
+                Write-Verbose "Creating UnitySetupInstance for $path"
+                [UnitySetupInstance]::new($path)
+            }
+            catch {
+                Write-Warning "$_"
+            }
         }
-        catch {
-            Write-Warning "$_"
-        }
-    }
 }
 
 <#

--- a/UnitySetup/UnitySetup.psm1
+++ b/UnitySetup/UnitySetup.psm1
@@ -1177,6 +1177,7 @@ function Get-UnitySetupInstanceVersion {
 
     Write-Verbose "Attempting to find UnityVersion in $path"
 
+    # Try to look in the modules.json file for installer paths that contain version info
     if ( Test-Path "$path\modules.json" -PathType Leaf ) {
 
         Write-Verbose "Searching $path\modules.json for module versions"
@@ -1191,9 +1192,10 @@ function Get-UnitySetupInstanceVersion {
         }
     }
 
+    # No version found, start digging deeper
     if ( Test-Path "$path\Editor" -PathType Container ) {
-        # We'll attempt to search for the version using the ivy.xml definitions for legacy editor compatibility.
-
+        
+        # Search for the version using the ivy.xml definitions for legacy editor compatibility.
         Write-Verbose "Looking for ivy.xml files under $path\Editor\"
         $ivyFiles = Get-ChildItem -Path "$path\Editor\" -Filter 'ivy.xml' -Recurse -ErrorAction SilentlyContinue -Force -File
         foreach ( $ivy in $ivyFiles) {
@@ -1203,11 +1205,22 @@ function Get-UnitySetupInstanceVersion {
 
             [xml]$xmlDoc = Get-Content $ivy.FullName
 
-            [string]$version = $xmlDoc.'ivy-module'.info.unityVersion
-            if ( -not $version ) { continue; }
+            [string]$ivyVersion = $xmlDoc.'ivy-module'.info.unityVersion
+            if ( -not $ivyVersion ) { continue; }
 
             Write-Verbose "`tFound version!"
-            return [UnityVersion]$version
+            return [UnityVersion]$ivyVersion
+        }
+
+        # Search through any header files which might define the unity version
+        Write-Verbose "Looking for .h files with UNITY_VERSION defined under $path\Editor\ "
+        $headerMatchInfo = Get-ChildItem -Path "$path\Editor\*.h" -Recurse -ErrorAction Ignore -Force -File | 
+            Select-String -Pattern "UNITY_VERSION\s`"(\d+\.\d+\.\d+[fpba]\d+)`"" | 
+            Select-Object -First 1
+
+        if ( $headerMatchInfo.Matches.Groups.Count -gt 1 ) {
+            Write-Verbose "`tFound version!"
+            return [UnityVersion]($headerMatchInfo.Matches.Groups[1].Value)
         }
     }
 }

